### PR TITLE
Extend PorscheDesignSystem{Provider|Module} to set default props on components, eg. dark theme (Issue/1185)

### LINKED ISSUE #1185: Extend PorscheDesignSystem{Provider|Module} to set default props on components, eg. dark theme
Scope
---
Since we are now in control of `PorscheDesignSystemProvider` in `components-{react|vue}` and `PorscheDesignSystemModule` in `components-angular` we are able to provide useful default configurations for our components that would otherwise be very repetitive. For example we could offer the ability to set the `theme` of all components within the `Provider|Module` to `dark|light|auto`.

Notes  
---
- Local component prop definition always has priority over a default configuration on Provider/Module level
- Ensure it's working in SSR context (NextJS/Remix) too
- What happens, when multiple Provider/Modules are getting initialised with different configurations?
- Provide a scalable solution to add more default props later on. Example how 
MaterialUI does it: https://mui.com/material-ui/customization/theme-components/
- Only possible for Frameworks (not for VanillaJS)

Acceptance criteria
---
- [x] A default theme (`dark|light|auto`) can be set on `Provider|Module` level for `components-{react|vue|angular}` which is getting reflected by all child components

Subtasks
---
- [x] Angular
- [x] React
- [x] React SSR
- [x] UXPin
- [x] Vue
- [x] Tests
- [x] Documentation
- [x] Changelog

### PR DESCRIPTION
**Pull request by denyo**
### Pull Request Checklist

- [x] I have read and accept the [Contributor License Agreement](https://opensource.porsche.com/docs/cla)

#### References

- Preview: 
- https://designsystem.porsche.com/issue/1185/developing/angular/advanced
- https://designsystem.porsche.com/issue/1185/developing/react/advanced
- https://designsystem.porsche.com/issue/1185/developing/vue/advanced

#### Scope

Short description of the pull request scope

#### Resolved Issue

Resolves #1185

### CODE REVIEW

File: packages/storefront/src/pages/developing/react/advanced.md [hunk 1]
```diff
@@ -2,6 +2,30 @@
...
+```tsx
+// index.tsx
+
+import ReactDOM from 'react-dom';
+import { PorscheDesignSystemProvider } from '@porsche-design-system/components-react';
+import { App } from './App';
+
+ReactDOM.render(
+  <React.StrictMode>
+    <PorscheDesignSystemProvider theme="dark">
```
Comment (stephanschroeter) [inline packages/storefront/src/pages/developing/react/advanced.md:21]: maybe add a comment which themes are possible `light | dark | auto (sync with operating system)`

File: packages/storefront/src/pages/developing/angular/getting-started.md [hunk 2]
```diff
@@ -49,16 +49,12 @@ Change your AppComponent to use at least one Porsche Design System component, fo
...
 @Component({
   selector: 'app-root',
-  template: `
-    <div id="app">
-      <p-heading>Welcome to Angular</p-heading>
-    </div>
-  `,
-  styles: [],
+  template: `<p-heading>Welcome to Angular</p-heading>`,
+  changeDetection: ChangeDetectionStrategy.OnPush,
```
Comment (stephanschroeter) [inline packages/storefront/src/pages/developing/angular/getting-started.md:57]: maybe add a comment why this is important?

File: packages/components-react/projects/react-wrapper/src/hooks.ts [hunk 3]
```diff
@@ -30,6 +30,14 @@ export const usePrefix = (tagName: string): string => {
   }
 };
 
+export const useTheme = (): Theme => {
+  if (process.env.NODE_ENV === 'test' && skipCheck) {
```
Comment (stephanschroeter) [inline packages/components-react/projects/react-wrapper/src/hooks.ts:34]: why is this necessary?

File: packages/components-js/tests/vrt/playwright/specs/states/button.vrt.ts [hunk 4]
```diff
@@ -43,7 +43,7 @@ const scenario = async (page: Page, theme: Theme, scheme?: PrefersColorScheme):
 };
 
 // executed in Chrome only
-test.describe(component, async () => {
+test.skip(component, async () => {
```
Comment (stephanschroeter) [inline packages/components-js/tests/vrt/playwright/specs/states/button.vrt.ts:46]: shouldn't be skipped I guess.

File: packages/components-js/public/index.html [hunk 5]
```diff
@@ -165,6 +165,7 @@
 
           app.innerHTML = '';
           app.innerHTML = (await (await fetch(`/${directory}/${page}.html`)).text()).replace(/>(\s)*</g, '><');
+          // TODO: why not replace all of this inside the static html instead of manipulating afterwards?
```
Comment (stephanschroeter) [inline packages/components-js/public/index.html:168]: I think that would be good.

File: packages/components-react/projects/react-wrapper/src/hooks.ts (hunk 3 above)
Comment (denyo) [reply packages/components-react/projects/react-wrapper/src/hooks.ts:34]: Because of the opt-out via `skipCheckForPorscheDesignSystemProviderDuringTests()`.  When this is called, there is no context and you will run into an exception.

File: packages/storefront/src/pages/developing/angular/advanced.md [hunk 6]
```diff
@@ -2,6 +2,61 @@
 
 <TableOfContents></TableOfContents>
 
+## Global Theme
+
+**Since v3.8.0** you can set the `theme` property for all child components by passing an option to
```
Comment (henrijoss) [inline packages/storefront/src/pages/developing/angular/advanced.md:7]: Shouldn't this be the next version released here? Since we are currently on 3.8.0

File: packages/components-angular/tests/e2e/specs/provider.e2e.ts [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+afterEach(async () => await page.close());
+
+it('should change p-button theme to dark', async () => {
+  await goto(page, 'button');
+
+  const button = await page.$('p-button');
+  const getTheme = (handle: ElementHandle<Element>): Promise<Theme> => handle.evaluate((el) => (el as any).theme);
+  expect(await getTheme(button)).toBe('light');
+
+  await page.select('select[name="theme"]', 'dark');
```
Comment (henrijoss) [inline packages/components-angular/tests/e2e/specs/provider.e2e.ts:16]: Since there are only three options, maybe test all of them here? Could also test the functionality in isolation since theres not guarantee the select is using the provider to set the theme

File: packages/components-js/tests/vrt/playwright/specs/common/components.vrt.ts [hunk 8]
```diff
@@ -128,6 +128,8 @@ components.forEach((component) => {
 
     // 200% font scaling
     test(`should have no visual regression for viewport ${baseViewportWidth} in scale mode`, async ({ page }) => {
+      test.skip(component === 'select-wrapper', 'select-wrapper is flaky');
```
Comment (henrijoss) [inline packages/components-js/tests/vrt/playwright/specs/common/components.vrt.ts:131]: Do we want to skip this test for now?

File: packages/components-angular/tests/e2e/specs/provider.e2e.ts (hunk 7 above)
Comment (denyo) [reply packages/components-angular/tests/e2e/specs/provider.e2e.ts:16]: I think testing every option doesn't bring additional value.  
How would you test the functionality in isolation?

File: packages/components-js/tests/vrt/playwright/specs/common/components.vrt.ts (hunk 8 above)
Comment (denyo) [reply packages/components-js/tests/vrt/playwright/specs/common/components.vrt.ts:132]: I'd say yes, better than retrying 10 times 😢

File: packages/components-angular/tests/e2e/specs/provider.e2e.ts (hunk 7 above)
Comment (henrijoss) [reply packages/components-angular/tests/e2e/specs/provider.e2e.ts:16]: Basically constructing the page within the test

**Review by henrijoss — APPROVED** [review]
Looks good to me
